pyclesperanto_prototype/_tier10/_to_igraph.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from .._tier0 import plugin_function
+from .._tier0 import Image
+
+def to_igraph(adjacency_matrix:Image, centroids:Image=None):
+    """
+    Converts a given adjacency matrix to a iGraph [1] graph data structure.
+
+    Note: the given centroids typically have one entry less than the adjacency matrix is wide, because
+    those matrices contain a first row and column representing background. When exporting the networkx graph,
+    that first column will be ignored.
+
+    Parameters
+    ----------
+    adjacency_matrix : Image
+        m*m touch-matrix, proximal-neighbor-matrix or n-nearest-neighbor-matrix
+    centroids : Image, optional
+        d*(m-1) matrix, position list of centroids
+
+    Returns
+    -------
+    iGraph graph
+
+    See Also
+    --------
+    ..[1] https://igraph.org/
+    """
+    try:
+        import igraph
+    except ImportError:
+        raise ImportError("igraph is not installed. Please refer to the documentation https://igraph.org/python/")
+
+    igraph_graph = igraph.Graph.Adjacency(np.asarray(adjacency_matrix)[1:,1:].tolist())
+
+    if centroids is not None:
+        from .._tier1 import transpose_xy
+        centroid_positions = transpose_xy(centroids)
+
+        for n in range(len(igraph_graph.vs)):
```

Comment:
You can iterate over the vertices directly:

```
for vertex in igraph_graph.vs:
    vertex["x"] = centroid_positions[n][0]
    ...etc...
```

However, it would be more efficient to construct lists of X, Y and Z coordinates for each vertex and then assign them to the graph in a single call (this saves you from constructing `Vertex` objects on the fly in the `for` loop just for the sake of assigning an attribute to them). Something like:

```
if centroids.shape[0] > 2:
    xs, ys, zs = [list(row) for row in centroid_positions.T]
else:
    xs, ys = [list(row) for row in centroid_positions.T]
    zs = [None] * igraph_graph.vcount()
igraph_graph.vs["x"] = xs
igraph_graph.vs["y"] = xs
igraph_graph.vs["z"] = zs
```